pkg/clustermanager/cluster_manager.go

```diff
@@ -342,12 +342,17 @@ func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, c
 }
 
 func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster, workloadCluster *types.Cluster, newClusterSpec *cluster.Spec, provider providers.Provider) error {
-	currentSpec, err := c.GetCurrentClusterSpec(ctx, workloadCluster, newClusterSpec.Cluster.Name)
+	eksaMgmtCluster := workloadCluster
+	if managementCluster != nil && managementCluster.ExistingManagement {
```

Comment:
Why do we need to do this here?
I'm probably missing something, but I would expect the workflow to pass the correct clusters, so ClusterManager doesn't need to to do any checking

Reply:
here we send the mgmt and workload clusters, but what we need inside `UpgradeCluster` are the clusters where we have capi resources and eksa crds. That is how these clusters are used in here